src/Controls/StreetControls.js

```diff
@@ -77,8 +79,143 @@ class StreetControls extends FirstPersonControls {
         this.previousPosition = newPreviousPosition;
     }
 
+    firstPickedObject(event) {
+        // basic threeJs pick with raycaster
+        const normalized = this.view.viewToNormalizedCoords(event);
+        this.raycaster.setFromCamera(normalized.clone(), this.camera);
+        const intersects = this.raycaster.intersectObjects(this.view.scene.children, true);
+
+        // skipping specific objects:
+        // - skip the two surfaces (rectangle and circle)
+        // - skip all the object without face, because we use the face to manage surface angle, so it's needed.
+        let i = 0;
+        while (i < intersects.length &&
+            (!intersects[i].face || intersects[i].object == this._rectangleMesh || intersects[i].object == this._circleMesh)) {
+            i++;
+        }
+
+        // return first interesting picked object, or undefined
+        if (i < intersects.length) {
+            return intersects[i];
+        }
+    }
+
+    pick(event, pickGround = () => {}, pickObject =  () => {}) {
+        // specific method to pick on the elevation layer at the good elevation, even if elevation layer is done in the shader
+        const mouse = this.view.eventToViewCoords(event);
+        const target = new THREE.Vector3();
+        if (this.view.getPickingPositionFromDepth(mouse, target)) {
+            // other way to pick object, with the basic threeJs raycast, and then objects are filtered.
+            const firstPicked = this.firstPickedObject(event);
+
+            // nothing is picked
+            if (!firstPicked) {
+                this.hideSurfaces();
+                return;
+            }
+
+            // get distance from the elevation layer
+            const distanceToGround = this.camera.position.distanceTo(target);
+            // compare first picked object distance to the distance of the elevation layer
+            if (!firstPicked.object.isTileMesh && firstPicked.distance < distanceToGround) {
+                // here we pick something, for example a building
+                target.copy(firstPicked.point);
+                // update recangle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._rectangleMesh, this._circleMesh);
+                // callback
+                pickObject(target);
+            } else {
+                // here we pick the ground, update circle surface helper
+                this.updateSurface(target, firstPicked.face.normal, this._circleMesh, this._rectangleMesh);
+                // callback
+                pickGround(target);
+            }
+        }
+    }
+
+    onClick() {
+        // here we detect mouse click when mouse is not moving,
+        // to avoid interfering with mouse drag movement, managed by FirstPersonControls
+        this.maybeClickWithoutMoving = true;
```

Comment:
This seems fishy: can you explain the interference you got ?

Reply:
I've already changed it.